### ISSUE TITLE
[apex] ApexCRUDViolationRule false-negative on non-VF getter

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3210]
**Affects PMD Version:** 
5.5.3 - 6.33

**Rule:**

[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

Currently (and since [5.5.3](https://github.com/pmd/pmd/commit/f9882bec87119e5c2e55e89c23272af9f524e796)) ApexCRUDViolation rule makes certain assumptions about a method name and signature to catalog it as a Visualforce getter, and therefore bypassing the checking for violations.

**Code Sample demonstrating the issue:**

Taken from the unit tests

```
public class Foo {
    public Contact getFoo() {
        String tempID = 'someID';
         Contact c = [SELECT Name FROM Contact WHERE Id=:tempID];
         return c;
    }
}
```

**Expected outcome:**

The above could, or could be not, a Visualforce related getter, there's currently no way to determine if a method is Visualforce related (contrary to how it is possible to determine with LWC and Aura with the `@AuraEnabled` annotation), someone could create a method that follows the current checking conditions and use it in any context therefore it shouldn't be considered as a VF getter.

For such use cases it should be advised for people to use the warning suppression capabilities or comment syntax for a particular method. 

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Apex PMD VSCode Extension
